core/services/flux_monitor.go

```diff
@@ -236,6 +243,51 @@ func (f pollingDeviationCheckerFactory) New(initr models.Initiator, runManager R
 	)
 }
 
+// ExtractFeedURLs extracts a list of url.URLs from the feeds parameter of the initiator params
+func ExtractFeedURLs(initr models.InitiatorParams, orm *orm.ORM) ([]*url.URL, error) {
+	var feeds []interface{}
+	var urls []*url.URL
+
+	err := json.Unmarshal(initr.Feeds.Bytes(), &feeds)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, entry := range feeds {
+		var bridgeURL *url.URL
+		var err error
+
+		switch entry.(type) {
+		case string: // feed url - ex: "http://example.com"
+			bridgeURL, err = url.ParseRequestURI(entry.(string))
+		case map[string]interface{}: // named feed - ex: {"bridge": "bridgeName"}
+			bridgeName := entry.(map[string]interface{})["bridge"].(string)
+			bridgeURL, err = GetBridgeURLFromName(bridgeName, orm) // XXX: currently an n query
```

Comment:
@se3000 can I get your opinion here... currently doing an n query for each named bridge in the list of feeds (for simplicity). I could change that but then preserving the order of the feeds would be a pain and would make the code more complex. Do you know if the order of the feeds matters? Does this optimization matter?

Reply:
ie - is it likely that we will have jobs with large numbers of named feeds? where doing a separate query for each would be costly? I'm guessing yes... so we should probably remove the n query. But that leads to my next question, which is.. does the order of the feeds in the array matter? Because if so it's going to take a bit of creativity to query for all the bridges at once and then re-construct the list of feeds in order.

Reply:
Ideally, order wouldn't matter because they'd all be fetched at the same time. I think this is good for now. We can optimize it in a later story when we start running into this as a problem.

Reply:
Another option is to add this as a method to the orm.   This seems pretty dependent on the database anyways.  It would be nice if it somehow passed which bridges are missing to the caller, but could be ugly.